internal/agent/collector/vsphere.go

```diff
@@ -159,7 +159,28 @@ func (c *Collector) run(ctx context.Context) {
 		return
 	}
 
-	zap.S().Named("collector").Infof("Create inventory")
+	zap.S().Named("collector").Infof("List Datastores")
+	datastores := &[]vspheremodel.Datastore{}
+	err = collector.DB().List(datastores, libmodel.FilterOptions{Detail: 1})
+	if err != nil {
+		zap.S().Named("collector").Errorf("failed to list datastores: %v", err)
+		return
+	}
+
+	zap.S().Named("collector").Infof("List Networks")
+	networks := &[]vspheremodel.Network{}
+	err = collector.DB().List(networks, libmodel.FilterOptions{Detail: 1})
+	if err != nil {
+		zap.S().Named("collector").Errorf("failed to list networks: %v", err)
+		return
+	}
+
+	zap.S().Named("collector").Infof("Extract cluster mapping and build helper maps")
+	clusterMapping, hostIDToPowerState, vmsByCluster := ExtractVSphereClusterIDMapping(*vms, *hosts, *clusters)
+	datastoreIndexToName, datastoreMapping := buildDatastoreMappings(*datastores)
+	networkMapping := buildNetworkMappingFromVSphere(*networks)
```

Comment:
I think we can improve performance and avoid having these loops here by changing "getDatastores" & "getNetworks" funcs (used in lines 186-187) to return map instead of array and than use it accordingly the change